brave/brave-core
# Added Shields domain-specific ad/fingerprint block setting questions

Resolves brave/brave-browser#21546

Resolves brave/brave-browser#22064
Resolves brave/brave-browser#22063

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

0. Open browser with default settings
1. Verify that the P3A metrics `Brave.Shields.DomainAdsSettingsAboveGlobal`, `Brave.Shields.DomainAdsSettingsBelowGlobal`, `Brave.Shields.DomainFingerprintSettingsAboveGlobal`, `Brave.Shields.DomainFingerprintSettingsBelowGlobal` all have values of 0 on the `brave://local-state` page.
2. Access a website, update the domain-specific ad-blocking setting to "Aggressive"
3. Verify that the P3A metric `Brave.Shields.DomainAdsSettingsAboveGlobal` has a value of 1.
4. Access another website, update the domain-specific ad-blocking setting to "Allow"
5. Verify that the P3A metric `Brave.Shields.DomainAdsSettingsBelowGlobal` has a value of 1, and that `Brave.Shields.DomainAdsSettingsAboveGlobal` still has a value of 1.
6. Access a website, update the domain-specific fingerprint blocking setting to "Aggressive"
7. Verify that the P3A metric `Brave.Shields.DomainFingerprintSettingsAboveGlobal` has a value of 1.
8. Access another website, update the domain-specific fingerprint blocking setting to "Allow"
9. Verify that the P3A metric `Brave.Shields.DomainFingerprintSettingsBelowGlobal` has a value of 1, and that `Brave.Shields.DomainFingerprintSettingsAboveGlobal` still has a value of 1.
